### PR TITLE
[GLUTEN-3408][CH] Remove ENABLE_LOCAL_FORMATS

### PR DESCRIPTION
**Pull request by baibaichen**
We intrduced `ENABLE_LOCAL_FORMATS` in https://github.com/oap-project/gluten/pull/1390, which make us cann't test our own parquet formt(defined in local engine) easily. This PR remove it and using runtime config instead of it.

1. For PARQUET, using Runtime Config(`spark.gluten.sql.columnar.backend.ch.runtime_config.use_local_format`) instead of it, upgrade ch_parquet to latest arrow.
2. For ORC, only use clickhouse format

Also, I Sync clang format from clickhouse, and update  IndentPPDirectives to None for better style

## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#3408)

## How was this patch tested?

using exited UT.

### CODE REVIEW

File: cpp-ch/local-engine/Storages/SubstraitSource/FormatFile.cpp [hunk 1]
```diff
@@ -77,15 +79,14 @@ FormatFilePtr FormatFileUtil::createFile(
 #if USE_PARQUET
     if (file.has_parquet())
     {
-        return std::make_shared<ParquetFormatFile>(context, file, read_buffer_builder);
+        bool useLocalFormat = context->getConfigRef().getBool("enable_local_parquet_format", false);
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Storages/SubstraitSource/FormatFile.cpp:82]: use_local_format

File: cpp-ch/local-engine/Storages/ch_parquet/CMakeLists.txt [hunk 2]
```diff
@@ -44,6 +44,7 @@ target_compile_options(ch_parquet PUBLIC -fPIC
     -Wno-suggest-override
     -Wno-unused-member-function
     -Wno-deprecated-this-capture
+    -Wno-tautological-unsigned-zero-compare
```
Comment (taiyang-li) [inline cpp-ch/local-engine/Storages/ch_parquet/CMakeLists.txt:47]: 加它是什么目的

File: backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetAQESuite.scala [hunk 3]
```diff
@@ -45,6 +45,9 @@ class GlutenClickHouseTPCHParquetAQESuite
       .set("spark.sql.autoBroadcastJoinThreshold", "10MB")
       .set("spark.gluten.sql.columnar.backend.ch.use.v2", "false")
       .set("spark.sql.adaptive.enabled", "true")
+      .set(
```
Comment (taiyang-li) [inline backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetAQESuite.scala:48]: 是否可同时覆盖开关的两个值。

**Review by taiyang-li — CHANGES_REQUESTED** [review]

File: cpp-ch/local-engine/Storages/ch_parquet/CMakeLists.txt (hunk 2 above)
Comment (baibaichen) [reply cpp-ch/local-engine/Storages/ch_parquet/CMakeLists.txt:47]: see `cpp-ch/local-engine/Storages/ch_parquet/arrow/encoding.cc#2888`, don't know why arrow try to test `unsigned int` is negitive or not

``` c++
uint32_t num_bytes = ..;
 // this is why need -Wno-tautological-unsigned-zero-compare 
 if (num_bytes < 0 || ...) {
  ..
 }
```

File: backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetAQESuite.scala (hunk 3 above)
Comment (baibaichen) [reply backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetAQESuite.scala:48]: All testcases use the same Parquet files, and only AQE test using local format.

**Review by taiyang-li — APPROVED** [review]
LGTM

**Review by zzcclp — APPROVED** [review]
LGTM

**Comment by baibaichen** [comment]
name | arrow(ms) | use_local_format(ms)| use_local_format-arrow
-- | -- | -- | --
q01 | 3849 | 3716 | -133
q02 | 3484 | 3186 | -298
q03 | 6031 | 5632 | -399
q04 | 3711 | 3557 | -154
q05 | 9845 | 9240 | -605
q06 | 1093 | 1268 | 175
q07 | 10854 | 10684 | -170
q08 | 4560 | 4672 | 112
q09 | 9228 | 9264 | 36
q10 | 6750 | 5917 | -833
q11 | 1852 | 1949 | 97
q12 | 4236 | 4062 | -174
q13 | 5043 | 3857 | -1186
q14 | 2097 | 2248 | 151
q15 | 3993 | 4198 | 205
q16 | 6168 | 6047 | -121
q17 | 10844 | 10909 | 65
q18 | 19723 | 18820 | -903
q19 | 3979 | 3594 | -385
q20 | 3292 | 3331 | 39
q22 | 1193 | 1154 | -39

**Pull request merged**